app/vmagent/remotewrite/remotewrite.go

```diff
@@ -491,13 +479,28 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, forceDropSamplesOnF
 		}
 		sortLabelsIfNeeded(tssBlock)
 		tssBlock = limitSeriesCardinality(tssBlock)
+		sas := sasGlobal.Load()
```

Comment:
Should we retrieve sas before `for len(tss) > 0 {` on L:448?

Reply:
moved it